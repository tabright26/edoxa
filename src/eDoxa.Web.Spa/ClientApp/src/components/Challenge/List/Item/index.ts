import ChallengeItem from "./Item";

export default ChallengeItem;
