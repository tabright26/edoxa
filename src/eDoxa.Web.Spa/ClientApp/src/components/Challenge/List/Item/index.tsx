import React, { FunctionComponent } from "react";
import { Card, CardBody, Row, Col } from "reactstrap";
import { LinkContainer } from "react-router-bootstrap";
import ChallengeLogo from "components/Challenge/Logo";
import ChallengeSummary from "components/Challenge/Summary";
import { getChallengeDetailsPath } from "utils/coreui/constants";
import { Challenge } from "types/challenges";

type Props = { challenge: Challenge };

const Item: FunctionComponent<Props> = ({ challenge }) => (
  <Card>
    <LinkContainer
      to={getChallengeDetailsPath(challenge.id)}
      style={{
        cursor: "pointer"
      }}
    >
      <CardBody>
        <Row>
          <Col md="3">
            <ChallengeLogo
              className="h-100 bg-gray-900"
              height={150}
              width={150}
            />
          </Col>
          <Col md="9">
            <ChallengeSummary challengeId={challenge.id} />
          </Col>
        </Row>
      </CardBody>
    </LinkContainer>
  </Card>
);

export default Item;
