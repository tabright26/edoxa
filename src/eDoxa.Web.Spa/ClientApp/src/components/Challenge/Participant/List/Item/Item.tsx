import React, { FunctionComponent, useState } from "react";
import { Collapse, Card, Badge, CardHeader, CardBody } from "reactstrap";
import Format from "components/Shared/Format";
import { ChallengeParticipant } from "types";
import ChallengeParticipantMatchList from "components/Challenge/Participant/Match/List";

interface Props {
  participant: ChallengeParticipant;
  position: number;
  payoutEntries: number;
}

const ChallengeParticipantItem: FunctionComponent<Props> = ({
  participant,
  position,
  payoutEntries
}) => {
  const [collapse, setCollapse] = useState(false);
  const toggle = () => setCollapse(!collapse);
  const doxatag = participant.user.doxatag
    ? participant.user.doxatag.name
    : "Data unavailable";
  return (
    <>
      <Card className="my-2" onClick={toggle}>
        <CardBody
          className={`p-0 d-flex ${
            collapse ? "bg-primary" : ""
          } border-0 rounded`}
        >
          <div
            className="pl-3 py-2 text-center"
            style={{
              width: "50px"
            }}
          >
            <Badge
              className={`${
                collapse
                  ? "bg-gray-700"
                  : position <= payoutEntries && participant.score
                  ? "bg-primary"
                  : "bg-secondary"
              } w-100`}
            >
              {collapse ? <strong>{position}</strong> : position}
            </Badge>
          </div>
          <div className="px-3 py-2">
            {collapse ? <strong>{doxatag}</strong> : doxatag}
          </div>
          <div
            className={`${
              collapse ? "bg-gray-700" : "bg-primary"
            } px-3 py-2 text-center ml-auto h-100 rounded-right`}
            style={{
              width: "125px"
            }}
          >
            <Format.Score score={participant.score} bold={collapse} />
          </div>
        </CardBody>
      </Card>
      <Collapse isOpen={collapse}>
        <Card>
          <CardHeader>Matches</CardHeader>
          <ChallengeParticipantMatchList participantId={participant.id} />
        </Card>
      </Collapse>
    </>
  );
};

export default ChallengeParticipantItem;
