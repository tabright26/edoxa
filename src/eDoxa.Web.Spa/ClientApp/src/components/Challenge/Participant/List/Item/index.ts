import ChallengeParticipantItem from "./Item";

export default ChallengeParticipantItem;
