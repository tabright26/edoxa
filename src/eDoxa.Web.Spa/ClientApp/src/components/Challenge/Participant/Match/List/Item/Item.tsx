import React, { FunctionComponent } from "react";
import { CardBody, Badge } from "reactstrap";
import Format from "components/Shared/Format";
import { compose } from "recompose";
import { ChallengeParticipantMatch } from "types";
import { MapDispatchToProps, connect } from "react-redux";
import { show } from "redux-modal";
import { CHALLENGE_MATCH_SCORE_MODAL } from "utils/modal/constants";

interface OwnProps {
  match: ChallengeParticipantMatch;
  position: number;
}

interface DispatchProps {
  showChallengeMatchScoreModal: () => void;
}

type InnerProps = DispatchProps;

type OutterProps = OwnProps;

type Props = InnerProps & OutterProps;

const ChallengeParticipantMatchItem: FunctionComponent<Props> = ({
  match,
  position,
  showChallengeMatchScoreModal
}) => (
  <CardBody className="p-0 border border-dark d-flex">
    <div
      className="pl-2 py-2 text-center"
      style={{
        width: "45px"
      }}
    >
      <Badge variant="light">{position}</Badge>
    </div>
    {/* <div
      className="px-3 py-2"
      style={{
        width: "350px"
      }}
    >
      <Moment unix format="LLLL">
        {match.synchronizedAt}
      </Moment>
    </div> */}
    <div
      className="py-2 text-center mx-auto"
      onClick={() => showChallengeMatchScoreModal()}
    >
      <Badge variant="primary">View details</Badge>
    </div>
    <div
      className="bg-primary px-3 py-2 text-center ml-5"
      style={{
        width: "125px"
      }}
    >
      <Format.Score score={match.score} />
    </div>
  </CardBody>
);

const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = (
  dispatch,
  ownProps
) => {
  return {
    showChallengeMatchScoreModal: () =>
      dispatch(
        show(CHALLENGE_MATCH_SCORE_MODAL, { stats: ownProps.match.stats })
      )
  };
};

const enhance = compose<InnerProps, OutterProps>(
  connect(null, mapDispatchToProps)
);

export default enhance(ChallengeParticipantMatchItem);
