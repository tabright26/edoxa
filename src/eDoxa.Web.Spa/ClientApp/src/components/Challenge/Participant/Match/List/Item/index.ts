import ChallengeParticipantMatchItem from "./Item";

export default ChallengeParticipantMatchItem;
