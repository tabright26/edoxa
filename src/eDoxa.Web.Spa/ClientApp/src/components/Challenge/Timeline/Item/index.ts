import ChallengeTimelineItem from "./Item";

export default ChallengeTimelineItem;
