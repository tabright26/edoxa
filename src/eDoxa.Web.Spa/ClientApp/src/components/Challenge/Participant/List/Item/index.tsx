import React, { FunctionComponent, useState } from "react";
import { Collapse, Card, Badge, CardHeader, CardBody } from "reactstrap";
import Format from "components/Shared/Format";
import { ChallengeParticipant } from "types";
import List from "components/Challenge/Participant/Match/List";
import { compose } from "recompose";
import {
  withUserProfileUserId,
  HocUserProfileUserIdStateProps
} from "utils/oidc/containers";

type Props = HocUserProfileUserIdStateProps & {
  participant: ChallengeParticipant;
  position: number;
  payoutEntries: number;
  bestOf: number;
};

const Item: FunctionComponent<Props> = ({
  participant,
  position,
  payoutEntries,
  bestOf,
  userId
}) => {
  const [collapse, setCollapse] = useState(false);
  const toggle = () => setCollapse(!collapse);
  const doxatag = participant.user.doxatag
    ? participant.user.doxatag.name
    : "Data unavailable";
  return (
    <>
      <Card className="my-2" onClick={toggle}>
        <CardBody
          className={`p-0 d-flex ${
            collapse ? "bg-primary" : ""
          } border-0 rounded`}
        >
          <div
            className="pl-3 py-2 text-center"
            style={{
              width: "50px"
            }}
          >
            <Badge
              className={`${
                collapse
                  ? "bg-gray-700"
                  : position <= payoutEntries && participant.score
                  ? "bg-primary"
                  : "bg-secondary"
              } w-100`}
            >
              {collapse ? <strong>{position}</strong> : position}
            </Badge>
          </div>
          <div
            className={`px-3 py-2 ${participant.user.id === userId &&
              "text-primary"}`}
          >
            {collapse ? (
              <strong className="text-light">{doxatag}</strong>
            ) : (
              doxatag
            )}
          </div>
          <div
            className={`${
              collapse ? "bg-gray-700" : "bg-primary"
            } px-3 py-2 text-center ml-auto h-100 rounded-right`}
            style={{
              width: "125px"
            }}
          >
            <Format.Score score={participant.score} bold={collapse} />
          </div>
        </CardBody>
      </Card>
      <Collapse isOpen={collapse}>
        <Card>
          <CardHeader className="d-flex">
            <strong className="text-uppercase my-auto">Match history</strong>
            <small className="ml-2 my-auto text-muted">
              ({participant.matches.length}/{bestOf})
            </small>
          </CardHeader>
          <List participantId={participant.id} />
        </Card>
      </Collapse>
    </>
  );
};

const enhance = compose<any, any>(withUserProfileUserId);

export default enhance(Item);
