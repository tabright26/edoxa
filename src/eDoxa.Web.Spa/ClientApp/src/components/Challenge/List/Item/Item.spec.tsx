import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import Item from "./Item";
import { ChallengesState } from "store/root/challenge/types";

it("renders without crashing", () => {
  //Arrange
  const challenges: ChallengesState = {
    data: [
      {
        id: "123",
        name: "",
        state: "Inscription",
        bestOf: 3,
        entries: 10,
        game: "LeagueOfLegends",
        payoutEntries: 5,
        entryFee: { currency: "token", amount: 0 },
        participants: [],
        timeline: {
          createdAt: 123123123,
          startedAt: null,
          endedAt: null,
          closedAt: null
        },
        scoring: new Map<string, string>(),
        payout: {
          prizePool: { currency: "token", amount: 200000 },
          buckets: []
        }
      },
      {
        id: "456",
        name: "",
        state: "Inscription",
        bestOf: 3,
        entries: 10,
        game: "LeagueOfLegends",
        payoutEntries: 5,
        entryFee: { currency: "token", amount: 0 },
        participants: [],
        timeline: {
          createdAt: 123123123,
          startedAt: null,
          endedAt: null,
          closedAt: null
        },
        scoring: new Map<string, string>(),
        payout: {
          prizePool: { currency: "token", amount: 200000 },
          buckets: []
        }
      },
      {
        id: "678",
        name: "",
        state: "Inscription",
        bestOf: 3,
        entries: 10,
        game: "LeagueOfLegends",
        payoutEntries: 5,
        entryFee: { currency: "token", amount: 0 },
        participants: [],
        timeline: {
          createdAt: 123123123,
          startedAt: null,
          endedAt: null,
          closedAt: null
        },
        scoring: new Map<string, string>(),
        payout: {
          prizePool: { currency: "token", amount: 200000 },
          buckets: []
        }
      }
    ],
    loading: false,
    error: null
  };

  const store: any = {
    getState: () => {
      return {
        root: {
          challenge: challenges
        }
      };
    },
    dispatch: action => {},
    subscribe: () => {}
  };

  const challenge = challenges.data.find(challenge => challenge.id === "123");

  //Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Item challenge={challenge} />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  //Assert
  expect(tree).toMatchSnapshot();
});
