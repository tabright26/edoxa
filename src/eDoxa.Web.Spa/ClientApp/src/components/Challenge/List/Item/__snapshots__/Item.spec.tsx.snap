// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="card-accent-primary card"
>
  <div
    className="card-header"
  >
    <div
      className="card-header-actions"
    >
      <span
        className="float-right badge badge-danger badge-pill"
      />
    </div>
  </div>
  <div
    className="card-body"
  >
    <div
      className="card-title"
    >
      Challenge Details
    </div>
    <p
      className="card-text"
    >
      This is a wider card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.
    </p>
    <p
      className="card-text"
    >
      <small
        className="text-muted"
      >
        <time
          dateTime={null}
        >
          Invalid date
        </time>
      </small>
    </p>
    <a
      aria-label={null}
      className="btn btn-primary"
      href="/arena/challenges/123"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      View Details
    </a>
  </div>
</div>
`;
